model/client4.go

```diff
@@ -2903,8 +2903,12 @@ func (c *Client4) DeletePost(postId string) (bool, *Response) {
 }
 
 // GetPostThread gets a post with all the other posts in the same thread.
-func (c *Client4) GetPostThread(postId string, etag string) (*PostList, *Response) {
-	r, err := c.DoApiGet(c.GetPostRoute(postId)+"/thread", etag)
+func (c *Client4) GetPostThread(postId string, etag string, collapsedThreads bool) (*PostList, *Response) {
```

Comment:
I understand it's a bit late but I'd like to point out that all these are effectively **breaking** changes to our Golang client API. 
I am curious to know if this was discussed before and accepted.

Reply:
@streamer45 - how do you figure we 'break' things? The additional parameter to GetPostThread function?

Reply:
Yes, any method/function signature change (as in changing the number of parameters or their type) would be breaking in Golang. This means that any client built on top of this will that makes use of those functions will fail to compile/execute as soon as its dependencies get updated. Usually breaking the API would require a major version bump (6.0 in our case). This kind of accidental breakage has happened before so it's not the end of the world but we should probably try to avoid this.

Reply:
@streamer45 - you are right. Thankfully this method is rather obscure and hopefully nobody outside of Mattermost is actually using it

Reply:
Well, this applies to all the affected methods so I wouldn't be surprised if someone is using one of them since they fetch posts (especially `GetPostsSince`). Maybe we could add a warning in the release notes. I think it would be better if it was a conscious decision rather than an overlook.